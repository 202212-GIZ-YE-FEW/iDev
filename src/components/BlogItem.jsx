import AOS from "aos";
import { getDownloadURL, ref } from "firebase/storage";
import Image from "next/image";
import Link from "next/link";
import { useTranslation, withTranslation } from "next-i18next";
import { useEffect, useState } from "react";

import "aos/dist/aos.css";

import Button from "@/components/ui/Button";

import { storage } from "@/firebase/config";
function BlogItem({ en_title, ar_title, id, isOdd, t }) {
    const [imageUrl, setImageUrl] = useState(null);
    const { i18n } = useTranslation();

    useEffect(() => {
        AOS.init({
            duration: 1000,
            delay: 500,
        });

        const imageRef = ref(storage, `blogImages/${id}`);
        getDownloadURL(imageRef)
            .then((url) => {
                setImageUrl(url);
            })
            .catch((error) => {
                // handle the error
            });
    }, [id]);
    return (
        <>
            <div className='relative mx-3'>
                <div className='h-60'>
                    <Image
                        src={
                            imageUrl ||
                            "https://developers.elementor.com/docs/assets/img/elementor-placeholder-image.png"
                        }
                        width={200}
                        height={140}
                        alt={`${en_title} thumbnail`}
                        sizes='(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw'
                        className='w-full h-full rounded-3xl'
                    />
                </div>
                <div className='absolute h-full w-full bg-black/70 text-yellow flex flex-col items-center justify-center bottom-0 opacity-0 rounded-3xl hover:bottom-0  hover:opacity-100 transition-all duration-1000'>
                    {/* <p data-aos='fade-top' className='mb-5'>
                        {i18n.language == "ar" ? `${ar_title}` : `${en_title}`}
                    </p> */}
                    <Link href={`/blogs/${id}`}>
                        <Button
                            content={t("more")}
                            text-transform='capitalize'
                            filled='false'
                            size='large'
                            fontSize='text-sm md:text-xl'
                            radius='md'
                        />
                    </Link>
                </div>

                {isOdd ? (
                    <div className='w-32 absolute bottom-2 left-[1rem]'>
                        <p className='uppercase text-base box-decoration-clone inline leading-7 bg-light-gray text-white px-2'>
                            {i18n.language == "ar"
                                ? `${ar_title}`
                                : `${en_title}`}
                        </p>
                    </div>
                ) : (
                    <p className='uppercase absolute top-2 left-2 text-base bg-light-gray text-white px-2'>
                        {i18n.language == "ar" ? `${ar_title}` : `${en_title}`}
                    </p>
                )}
            </div>
        </>
    );
}
export default withTranslation("common")(BlogItem);
