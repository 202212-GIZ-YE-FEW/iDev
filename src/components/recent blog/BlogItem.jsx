import Image from "next/image";
import { withTranslation } from "next-i18next";

function BlogItem({ title, thumbnail, isOdd }) {
    return (
        <>
            <div className='relative'>
                <Image
                    src={`/home/${thumbnail}.svg`}
                    alt={`${title} svg`}
                    width={200}
                    height={160}
                    sizes='(max-width: 768px) 100vw,
                        (max-width: 1200px) 50vw,
                        33vw'
                    style={{ height: "100%", width: "100%" }}
                />
                {isOdd ? (
                    <div className='w-32 absolute bottom-2 left-2'>
                        <p className='uppercase text-base box-decoration-clone inline leading-7 bg-light-gray text-white px-2'>
                            {title}
                        </p>
                    </div>
                ) : (
                    <p className='uppercase absolute top-2 left-2 text-base text-white px-2'>
                        {title}
                    </p>
                )}
            </div>
        </>
    );
}
export default withTranslation("home")(BlogItem);
