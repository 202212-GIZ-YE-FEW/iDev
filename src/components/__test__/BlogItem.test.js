import renderer from "react-test-renderer";

import BlogItem from "../BlogItem";

it("renders correctly", () => {
    const tree = renderer.create(<BlogItem />).toJSON();
    expect(tree).toMatchSnapshot();
});
