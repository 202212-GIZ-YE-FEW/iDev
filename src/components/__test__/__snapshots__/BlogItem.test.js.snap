// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative mx-3"
>
  <div
    className="h-60"
  >
    <img
      alt="undefined thumbnail"
      className="w-full h-full rounded-3xl"
      data-nimg="1"
      decoding="async"
      height={140}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
      src="/_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=3840&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fdevelopers.elementor.com%2Fdocs%2Fassets%2Fimg%2Felementor-placeholder-image.png&w=3840&q=75 3840w"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={200}
    />
  </div>
  <div
    className="absolute h-full w-full bg-black/70 text-yellow flex flex-col items-center justify-center bottom-0 opacity-0 rounded-3xl hover:bottom-0  hover:opacity-100 transition-all duration-1000"
  >
    <a
      href="/blogs/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="transform hover:bg-yellow transition duration-500  hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  capitalize text-sm md:text-xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium border-2 border-cyan text-cyan"
        disabled={false}
      >
        more
      </button>
    </a>
  </div>
  <p
    className="uppercase absolute top-2 left-2 text-base bg-light-gray text-white px-2"
  >
    undefined
  </p>
</div>
`;
